### PR TITLE
chore(test): vitest 導入と app/lib Pure関数の最小ユニットテスト71件

### PR DESCRIPTION
**Pull request by igomuni**
## 目的

app/lib/ の Pure 関数群（クエリ・検索・highlights・レイアウト・エージェント）が自動テストゼロのまま積み上がり、品質の守りが tsc/lint/手動ハーネスのみである状態を解消するため（`docs/tasks/20260710_2324_技術的懸念の棚卸し` 懸念1・承認済み優先①）。

## 変更内容

- **vitest 導入**（node 環境のみ・React テストはスコープ外）: `vitest.config.ts`（`@/*` エイリアス解決）+ `npm test` / `npm run test:watch`
- **8モジュール・71テスト**（実行約250ms）。テストは対象モジュールの隣に併置:

| モジュール | 件数 | 重点 |
|-----------|-----:|------|
| sankey-query | 17 | resolve検証・URLラウンドトリップ・topShare 0除算→null・compareYears diffRate null |
| subcontract-layout | 14 | 深度サイクル/上限・note結合・**バンド不変条件**（子は親バンド内・重なりなし・単一子は同中心） |
| project-search | 10 | normalizeQuery・scope=details の matchedIn 優先 |
| highlights | 9 | nearest-rank分位点・10億円下限・multiSignal 共起 |
| spending-search | 8 | 直接/再委託の分離集計・excerpt 位置写像（空白/NFKC） |
| ribbon-layout | 5 | **入口整合±0.5px・部分再委託（Σ出口<バー高さ）**・#264レビュー修正の回帰 |
| font-scale | 5 | 丸め・基準値で恒等 |
| sankey-chat-agent | 3 | モック LlmCaller で終了3経路（テキスト/確定/ツール上限） |

- **実データ非依存**: すべて合成フィクスチャ。public/data 未展開の環境・CIでもそのまま走る（fs を触るローダ群は対象外）
- `/quality-check` スキルに `npm test` を追加 — 以後の全PRでテスト実行が標準に
- 境界値テストで**既存バグの発見なし**（0除算・分位点・圧縮フォールバック等すべて仕様どおり）

## テスト方法

```bash
npm test          # 71 passed, ~250ms
npx tsc --noEmit  # クリーン
npm run build     # 影響なし（テストがルートに漏れないことを確認済み）
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added Vitest-based unit test commands, including standard and watch modes.
  * Added automated coverage for chat agents, search, Sankey analysis, highlighting, font scaling, and subcontract layouts.
  * Added a documented quality-check step requiring unit tests alongside linting and type checking.
* **Tests**
  * Added validation for query handling, calculations, filtering, sorting, pagination, layout constraints, and edge cases.

### CODE REVIEW

File: app/lib/highlights.test.ts [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+      qualityItems,
+    });
+    const pids = result.metrics.execBudgetGap.map(e => e.pid);
+    expect(pids).toEqual(['3']);
+  });
+
+  it('execBudgetGap excludes items with null execAmount', () => {
+    const qualityItems: QualityScoreItem[] = [
+      makeQualityItem({ pid: '1', name: '執行額未定', budgetAmount: 100, execAmount: null as unknown as number }),
+    ];
```
[the remark this answers is not in the document]
Comment (igomuni) [reply app/lib/highlights.test.ts:150]: 対応しました: execAmount を number|null に修正（生JSONでnullになりうる・highlights.ts もガード済みで型が実行時と矛盾していました）。併せて formatAmount を null 許容にし、execAmount=null 時に formatAmount 内でクラッシュしうる潜在バグ（quality/page・ScoreDetailDialog）も解消しています。
